### DISCUSSION TITLE
Install directus on different schema.

### DISCUSSION THREAD

**Discussion by JonL1** [directus/directus#10453] [Archive]
Can you install directus on a specific schema('directus') different from your main app DB('public')?

**Comment by ilijaNL** [comment]
If you using postgres, set env: DB_SEARCH_PATH: 'content'

**Reply by JonL1** [reply]
I thought that variable was so that Directus could interface with a different DB schema where your content lays.
What I'm asking is to install Directus in a different schema other than 'public'. i.e. 'directus' schema

**Reply by ilijaNL** [reply]
> I thought that variable was so that Directus could interface with a different DB schema where your content lays. What I'm asking is to install Directus in a different schema other than 'public'. i.e. 'directus' schema

If you just specify one search path schema, the system tables of directus will be installed on that schema. After you can extend this environent with multiple schemas such that directus will scan all defined schemas for content. This is however cumbersome for now

**Reply by JonL1** [reply]
Thanks for the extended explanation. Not ideal then.
I was hoping that the core of the database layer used Knex withSchema function to perform DB operations.

**Reply by ilijaNL** [reply]
I don't think directus wil support this since not all SQL database have a concept of schemas and directus is db agnostic, at least that is what they say @rijkvanzanten

**Reply by rijkvanzanten** [reply]
@ilijaNL that's correct. More importantly, knex's `withSchema` has to be added every query, rather than being a global configuration option, and it doesn't gracefully degrade when schema's aren't supported, so that takes a bit more effort than just adding a single method call.

**Reply by JonL1** [reply]
Thanks both for the replies! Sometimes I forget that PostgreSQL is not the only DB out there :)

**Comment by JonL1** [comment]
> I don't think directus wil support this since not all SQL database have a concept of schemas and directus is db agnostic

> More importantly, knex's withSchema has to be added every query, rather than being a global configuration option, and it doesn't gracefully degrade when schema's aren't supported, so that takes a bit more effort than just adding a single method call.

**Comment by azrikahar** [comment]
potential workaround for this is shared over here: #12057
